addon_imps/citations/mendeley.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+
+from addon_toolkit.interfaces.citation import (
+    CitationAddonImp,
+    ItemResult,
+    ItemSampleResult,
+    ItemType,
+)
+
+
+class MendeleyCitationImp(CitationAddonImp):
+    async def get_external_account_id(self, auth_result_extras: dict[str, str]) -> str:
+        async with self.network.GET(
+            "profiles/me",
+        ) as response:
+            profile_info = await response.json_content()
+            user_id = profile_info.get("id")
+            if not user_id:
+                raise KeyError("Failed to fetch user ID from Mendeley.")
+            return str(user_id)
+
+    async def list_root_collections(self) -> ItemSampleResult:
+        async with self.network.GET("folders") as response:
+            response_json = await response.json_content()
+            return self._parse_collection_response(response_json)
+
+    async def list_collection_items(
+        self,
+        collection_id: str,
+        filter_items: ItemType | None = None,
+    ) -> ItemSampleResult:
+        async with self.network.GET(
+            f"folders/{collection_id}/documents",
+        ) as response:
+            document_ids = await response.json_content()
+            items = await self._fetch_documents_details(document_ids, filter_items)
+
+            return ItemSampleResult(items=items, total_count=len(items))
+
+    async def _fetch_documents_details(
+        self, document_ids: list[dict], filter_items: ItemType | None
+    ) -> list[ItemResult]:
+        items = []
+
+        tasks = [
+            self.fetch_item_details(doc["id"], filter_items, items)
+            for doc in document_ids
+        ]
+        await asyncio.gather(*tasks)
+
+        return items
```

Comment:
please rewrite this. `asyncio.gather` returns results of all of it's coroutines in a tuple